File: Practica_3/java/Pila.java
package Practica_3.java;

public class Pila {
    private long arreglo[];
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        arreglo = new long[n];
        top = -1;
    }
    public void push(int e) {
        if (isFull()) {
            System.out.println("Pila llena");
        } else {
            top++;
            arreglo[top] = e;
            
        }
    }
    public long pop(){
        if (isEmpty()) {
            System.out.println("Pila vacia");
            return -1;
        } else {
            long e = arreglo[top];
            top--;
            return e;
        }
    }
    public long peek() {
        return arreglo[top];
        }
    public boolean isEmpty() {
        return (top == -1);
    }
    public boolean isFull() {
        return (n-1 == top);
    }

}

File: Practica_3/python/Pila.py
class Pila:
    def __init__(self, n):
        self.__arreglo = [0] * n
        self.__top = -1
        self.__n = n

    def push(self, e):
        if self.is_full():
            print("Pila llena")
        else:
            self.__top += 1
            self.__arreglo[self.__top] = e

    def pop(self):
        if self.is_empty():
            print("Pila vacía")
            return -1
        else:
            e = self.__arreglo[self.__top]
            self.__top -= 1
            return e

    def peek(self):
        return self.__arreglo[self.__top]

    def is_empty(self):
        return self.__top == -1

    def is_full(self):
        return self.__top == self.__n - 1

